### PR TITLE
docs/vault-secrets-operator: GKE workload identity auth

### PR DESCRIPTION
**Pull request by tvoran**
Adding overview docs for using GKE workload identity with Vault Secrets Operator under `Secret Sources/Vault/Auth Methods/`. Updates the Vault Auth method section in the `Vault/Auth Methods` overview page with links to the VSO API sections for the other supported auth methods (until they have their own pages).

Implemented in https://github.com/hashicorp/vault-secrets-operator/pull/411

Docs preview: 
- https://vault-a78r8ypve-hashicorp.vercel.app/vault/docs/platform/k8s/vso/sources/vault#supported-vault-authentication-methods
- https://vault-a78r8ypve-hashicorp.vercel.app/vault/docs/platform/k8s/vso/sources/vault/gcp-auth

### CODE REVIEW

File: website/content/docs/platform/k8s/vso/auth/gcp.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+spec:
+  vaultConnectionRef: <VAULT_CONNECTION_NAME>
+  mount: gcp
+  method: gcp
+  gcp:
+    role: <VAULT_GCP_ROLE>
+    workloadIdentityServiceAccount: <SERVICE_ACCOUNT>
+```
+
+See the full list of gcp auth options on the [api page](https://github.com/hashicorp/vault-secrets-operator/blob/c9eb11f776c7f86895383fc32740bbb73e141aa5/docs/api/api-reference.md#vaultauthconfiggcp).
```
Comment (tomhjp) [inline website/content/docs/platform/k8s/vso/auth/gcp.mdx:67]: Should this link to the API reference docs instead? I don't see them added to this PR yet though, so perhaps I'm missing something.

**Review by tomhjp — COMMENTED** [review]
LGTM - the only major question I have is whether we should have updated API reference docs included in this PR?

File: website/content/docs/platform/k8s/vso/auth/gcp.mdx (hunk 1 above)
Comment (tvoran) [reply website/content/docs/platform/k8s/vso/auth/gcp.mdx:67]: Oh yep, for sure. The API docs will go out as part of the operator release, so we can add the real link then.

**Review by tomhjp — APPROVED** [review]

File: website/content/docs/platform/k8s/vso/auth/gcp.mdx [hunk 2]
```diff
@@ -9,7 +9,7 @@ description: >-
 
 The Vault Secrets Operator (VSO) supports authenticating to Vault's [GCP auth](/vault/docs/auth/gcp) method, using GKE workload identity.
 
-See Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) for the steps necessary to enable workload identity on a GKE cluster, and to allow a Kubernetes service account to impersonate a Google IAM service account.
+See Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) for the steps necessary to enable workload identity on a GKE cluster, and to allow a Kubernetes service account to impersonate a Google IAM service account. The [workload identity submodule](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/28.0.0/submodules/workload-identity) of the [GKE terraform module](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/28.0.0) can be used for this setup as well.
```
Comment (benashz) [inline website/content/docs/platform/k8s/vso/auth/gcp.mdx:12]: Should we use the latest version, instead of pinning?

Comment (tvoran) [reply website/content/docs/platform/k8s/vso/auth/gcp.mdx]: Oops, yep.

File: website/content/docs/platform/k8s/vso/auth/gcp.mdx
Comment (benashz) [inline website/content/docs/platform/k8s/vso/auth/gcp.mdx:1]: I think this probably should go under https://developer.hashicorp.com/vault/docs/platform/k8s/vso/sources/vault

**Review by benashz — COMMENTED** [review]
We re-jigged the docs layout when we added support for HCPVS.  I think we should probably move this to be under https://developer.hashicorp.com/vault/docs/platform/k8s/vso/sources/vault

**Comment by tvoran** [comment]
Rearranged this similar to #23982

File: website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+page_title: GCP auth support for Vault Secrets Operator
+description: >-
+  Learn how GCP authentication works for Vault Secrets Operator
+---
+
+# GCP auth support for Vault Secrets Operator
+
+The Vault Secrets Operator (VSO) supports authenticating to Vault's [GCP auth](/vault/docs/auth/gcp) method, using Google's Kubernetes Engine (GKE) workload identity.
+
+1. Follow Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to enable workload identity on a GKE cluster so your Kubernetes service account can impersonate a Google IAM service account.
```
Comment (tomhjp) [inline website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx:12]:
```suggestion
1. Follow Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) guide to enable workload identity on a GKE cluster so your Kubernetes service account can impersonate a Google IAM service account. 
```

File: website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  metadata:
+    name: vaultauth-gcp-example
+    namespace: <K8S_NAMESPACE>
+  spec:
+    vaultConnectionRef: <VAULT_CONNECTION_NAME>
+    mount: gcp
+    method: gcp
+    gcp:
+      role: <VAULT_GCP_ROLE>
+      workloadIdentityServiceAccount: <SERVICE_ACCOUNT>
```
Comment (tomhjp) [inline website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx:66]: Is it required to be in exactly this form, or do we also support it with the `@<GCP_PROJECT>.iam.gserviceaccount.com` suffix like it appears above? Reading the page with fresh eyes, it's not totally obvious that the suffix was intentionally omitted. I kind of wish we had the ability to interactively substitute in real values like Google's docs can.

**Review by tomhjp — APPROVED** [review]
LGTM! I'm not sure how to improve on my minor point of confusion, but it's very minor anyway.

File: website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx (hunk 4 above)
Comment (tvoran) [reply website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx:66]: Oh good question, this is the k8s service account, which often is just be the `<SERVICE_ACCOUNT>` portion of the GCP service account, but I'll change this to `<K8S_SERVICE_ACCOUNT>` to make it more clear which is expected here.

And yeah, +1 to Google's doc substitution, it's really nice for things like this.

**Review by benashz — COMMENTED** [review]
LGTM!

**Review by benashz — APPROVED** [review]

**Comment by tvoran** [comment]
Thanks!

**Pull request merged**